examples/file-sharing.rs

```diff
@@ -134,8 +136,17 @@ async fn main() -> Result<(), Box<dyn Error>> {
                     // Reply with the content of the file on incoming requests.
                     Some(network::Event::InboundRequest { request, channel }) => {
                         if request == name {
-                            let file_content = std::fs::read_to_string(&path)?;
-                            network_client.respond_file(file_content, channel).await;
+                            let mut reader = BufReader::new(File::open(&path)?);
+                            let mut vec = Vec::<u8>::new();
+                            let mut buffer = reader.fill_buf()?;
+                            let mut len = buffer.len();
+                            while len > 0 {
+                                vec.extend_from_slice(buffer);
+                                reader.consume(len);
+                                buffer = reader.fill_buf()?;
+                                len = buffer.len();
+                            }
```

Comment:
Given that this is reading the entire file into `vec`, what is the difference to the previous version, i.e. `std::fs::read_to_string`? If you want to read an entire file as `u8`, why not use the below?

https://doc.rust-lang.org/std/fs/fn.read.html

Reply:
OK

Reply:
OK. fixed.